src/strokes.ts

```diff
@@ -207,36 +179,65 @@ export class Stroke extends Modifier {
         throw new RuntimeError('InvalidType', `The stroke type ${this.type} does not exist`);
     }
 
-    let strokeLine = 'straight';
     // Draw the stroke
     if (this.type === Stroke.Type.BRUSH_DOWN || this.type === Stroke.Type.BRUSH_UP) {
       ctx.fillRect(x + this.x_shift, topY, 1, botY - topY);
+      // Draw the arrow head
+      Glyph.renderGlyph(
+        ctx,
+        x + this.x_shift - Glyph.getWidth(arrow, this.render_options.font_scale) / 2,
+        arrow_y,
+        this.render_options.font_scale,
+        arrow
+      );
     } else {
-      strokeLine = 'wiggly';
-      if (isStaveNote(note)) {
-        for (let i = topY; i <= botY; i += line_space) {
-          Glyph.renderGlyph(ctx, x + this.x_shift - 4, i, this.render_options.font_scale, 'vexWiggleArpeggioUp');
-        }
-      } else {
-        let i;
-        for (i = topY; i <= botY; i += 10) {
-          Glyph.renderGlyph(ctx, x + this.x_shift - 4, i, this.render_options.font_scale, 'vexWiggleArpeggioUp');
-        }
-        if (this.type === Stroke.Type.RASQUEDO_DOWN) {
-          text_y = i + 0.25 * line_space;
-        }
+      const h = Glyph.getWidth(
+        arrow == 'arrowheadBlackDown' ? 'wiggleArpeggiatoDown' : 'wiggleArpeggiatoUp',
+        this.render_options.font_scale
+      );
+      ctx.openRotation(90, x + this.x_shift - h / 4, topY - (h * 3) / 4);
+      let i = 0;
+      if (arrow == 'arrowheadBlackUp') {
+        // Draw the arrow head
+        Glyph.renderGlyph(
+          ctx,
+          x + this.x_shift - h / 4 - 4,
+          topY -
+            (h * 3) / 4 -
+            Glyph.getHeight('wiggleArpeggiatoUp', this.render_options.font_scale) / 2 +
+            Glyph.getHeight('arrowheadBlackLeft', this.render_options.font_scale) / 2,
+          this.render_options.font_scale,
+          'arrowheadBlackLeft'
+        );
       }
+      for (; i <= botY - topY; i += 10) {
+        Glyph.renderGlyph(
+          ctx,
+          x + this.x_shift - h / 4 + i,
+          topY - (h * 3) / 4,
+          this.render_options.font_scale,
+          arrow == 'arrowheadBlackDown' ? 'wiggleArpeggiatoDown' : 'wiggleArpeggiatoUp'
+        );
+      }
+      if (arrow == 'arrowheadBlackDown') {
+        // Draw the arrow head
+        Glyph.renderGlyph(
+          ctx,
+          x + this.x_shift - h / 4 + i - 10 + 4,
```

Comment:
Can you explain some of these calculations, and also in some of the blocks above as well? Someone looking at this code in the future won't be able to know exactly what we are trying to accomplish here.

I just want to make sure the arrow heads are centered horizontally in all of our fonts. Right now some of them don't look centered to me. 

Can you generate zoomed in images with the wiggle + arrowhead for all fonts? We need to position the arrow head so it looks centered (and touching the wiggle at the narrowest point).

Reply:
Code reworked to make it easier to understand and easier to comment